src/python/pants/option/options.py

```diff
@@ -499,8 +499,7 @@ def log(log_func):
                 raise Exception(
                     "Option value mismatches detected, see logs above for details. Aborting."
                 )
-        # TODO: In a future release, switch to the native_values as authoritative.
-        return legacy_values
+        return native_values
```

Comment:
I wonder if we want to allow a user to fallback to the legacy values if things are wildly wrong. E.g. `[GLOBAL].use_deprecated_legacy_options_values = true` or something. Thoughts?

(Happy to get this PR merged ASAP, as is, and we can cherry-pick any tweaks along these back to 2.23.x if we decide on them)

Reply:
I don't think we do - 2.22.x is where you're supposed to discover if things are wildly wrong.